zmq-lwt/src/zmq_lwt.ml

```diff
@@ -0,0 +1,82 @@
+let (>>=) = Lwt.(>>=)
+
+module Socket = struct
+
+  type 'a t = {
+    socket : 'a ZMQ.Socket.t;
+    fd : Lwt_unix.file_descr;
+  }
+
+  exception Break_event_loop
+
+  let of_socket socket = {
+    socket;
+    fd = Lwt_unix.of_unix_file_descr ~blocking:false ~set_flags:false (ZMQ.Socket.get_fd socket);
+  }
+
+  let to_socket s = s.socket
+
+  (* Wrap possible exceptions and events which can occur in a ZeroMQ call *)
+  let wrap f s =
+    let io_loop () =
+      Lwt_unix.wrap_syscall Lwt_unix.Read s.fd (
```

Comment:
By reading the documentation on wrap_syscall, lwt will monitor the given fd and only call the given function when fd becomes readable/writable based on the argument given.

In the read case I think this is ok to just use read, but for the write case, I would fear that we could block forever if we never receive any messages. I dont think its pattern can be used for zmq, as its meant for simple unix writes (which we re not doing) and not for use with an fd. 

Also the zmq documentation says that fd is edge triggered, and requires an call to Zmq.Socket.events before we wait on the fd.